Mathlib/Analysis/Convex/Basic.lean

```diff
@@ -595,3 +601,44 @@ protected theorem starConvex (K : Submodule 𝕜 E) : StarConvex 𝕜 (0 : E) K
   K.convex K.zero_mem
 
 end Submodule
+
+section CommSemiring
+
+variable {R : Type*} [CommSemiring R]
+variable (A : Type*) [Semiring A] [Algebra R A]
+variable {M : Type*} [AddCommMonoid M] [Module A M] [Module R M] [IsScalarTower R A M]
+variable [PartialOrder R] [PartialOrder A]
+
+lemma convex_of_nonneg_algebraMap {s : Set M} (halg : ∀ ⦃r : R⦄, 0 ≤ r → 0 ≤ algebraMap R A r)
+    (hs : Convex A s) : Convex R s := by
+  apply Convex.lift_of_smul_of_nonneg_right R hs
+  intro r hr
+  rw [← Algebra.algebraMap_eq_smul_one]
+  exact halg hr
+
+lemma convex_of_nonneg_surjective_algebraMap [FaithfulSMul R A] {s : Set M}
+    (halg : ∀ ⦃a : A⦄, 0 ≤ a → ∃ (r : R), 0 ≤ r ∧ algebraMap R A r = a) (hs : Convex R s) :
```

Comment:
This condition `halg` can be written as `Set.Ici 0 ⊆ algebraMap R A '' Set.Ici 0`, which seems more readable.